### ISSUE TITLE
[FEATURE] Support third-party Discord servers

### ISSUE THREAD

**Issue by erkinalp** [luphoria/graim#9]
Third party Discord servers such as https://gitlab.com/litecord/litecord and https://github.com/fosscord/fosscord exist. It would be great to be able to use this bot with those third party servers.

**Comment by luphoria** [comment]
Have you tried supporting Litecord? As just an API implementation it should work.

Anyhow - with Fosscord - i'm looking into it, but it looks like it may be a similar situation (we just would have to wait for them to finish server implementation to see). However, I don't plan on "officially" supporting Fosscord, mostly for a few things that are sort of scaring me away:

## Fosscord claims "end to end encryption"
![image](https://user-images.githubusercontent.com/60309933/167152494-3db1c44e-8b1c-4dbd-a244-5153810a89a6.png)
The issue with this statement, of course, is that e2ee is not in their codebase. There is only one mention of encryption in the entire client codebase, and searching the entire server repository, only [this commit](https://github.com/fosscord/fosscord-server/commit/31bfe2a22a81b0878e9232d91b19925ff12d4007) references any work on E2EE. A bold claim like this is an issue in my eyes.

## Fosscord claims "we don't make money"
![image](https://user-images.githubusercontent.com/60309933/167153342-863d2361-84a6-49d4-b2cc-6dafb751cd19.png)
But - surprise - they have a large number of financial contributors, and their [OpenCollective page](https://opencollective.com/fosscord) displays quite a hefty sum..

![image](https://user-images.githubusercontent.com/60309933/167153656-4499ddef-5bea-47ab-90b5-16e6581fec77.png)
## Fosscord rips Discord-copyright material
This one is pretty obvious, but their client (despite being described as its own thing) is that it pretty much just rips Discord, and the security really just isn't there yet.

![image](https://user-images.githubusercontent.com/60309933/167154569-ea233344-a885-4aca-8838-cd109b75b21c.png)
Take this screenshot - the client still requests discord.com for status. 

![image](https://user-images.githubusercontent.com/60309933/167154678-4a52fedf-f0e3-4774-b232-dac3e8bbe5c8.png)
It clearly views itself as just a Discord decompilation of sorts. Just about every hyperlink redirects you to Discord, and in fact, there isn't a single time it calls itself fosscord.

TL;DR: I understand the want for support for something like this, but it really isn't something I plan on. In the further future, I may consider making graim 1:1 with a Matrix bridge "swiss knife" if one exists, but something like Fosscord really isn't on my radar.

**Comment by erkinalp** [comment]
> just a Discord decompilation of sorts

Hello, I am the maintainer of fosscord. We do not decompile/rip Discord client. What actually happens in demo instances is the Discord client is locally fetched in the user's browser, but with the API endpoints pointed to the respective fosscord instance instead of discord.com.

https://github.com/fosscord/fosscord-server/blob/bc76c840b805d52a086f05dd4ea7e996a8abf2a0/api/client_test/index.html

**Comment by luphoria** [comment]
> > just a Discord decompilation of sorts
> 
> Hello, I am the maintainer of fosscord. We do not decompile/rip Discord client. What actually happens in demo instances is the Discord client is locally fetched in the user's browser, but with the API endpoints pointed to the respective fosscord instance instead of discord.com.
> 
> https://github.com/fosscord/fosscord-server/blob/bc76c840b805d52a086f05dd4ea7e996a8abf2a0/api/client_test/index.html

Yeah, i understand this. If the API is 1:1 (should be, yes?), then graim should be compatible :P

Hope you fix up the stuff I mentioned, though. Good luck with your project
